src/messages/MOSDECSubOpRead.h

```diff
@@ -20,8 +20,8 @@
 
 class MOSDECSubOpRead : public MOSDFastDispatchOp {
 private:
-  static constexpr int HEAD_VERSION = 3;
-  static constexpr int COMPAT_VERSION = 1;
+  static constexpr int HEAD_VERSION = 4;
+  static constexpr int COMPAT_VERSION = 2;
```

Comment:
why did you change the `COMPAT_VERSION`?

Reply:
because the change to the message structure is not just adding something at the end of the msg. according to https://docs.ceph.com/en/latest/dev/encoding/

Reply:
you can't just bump compat_version by 1, you'd have to set it to the minimum version you can still decode. if you remove stuff in the v4 encoding, then compat_version also needs to be 4. but that means we'd break all backward compatibility with existing clients

so we need to do this without changing compat versions. unfortunately i think that means we still need to call `encode_trace()`/`decode_trace()` even if we're going to ignores its values

Reply:
i dont think we need to change the compatibility version at all. we kept the blkin info struct and we are decoding it

Reply:
thanks for the clarification! will fix it

Reply:
> i dont think we need to change the compatibility version at all. we kept the blkin info struct and we are decoding it

if we remove the `encode_trace()` part from the v4 encoding, then receivers still running v3 would fail to decode

Reply:
ok. so, should we just keep that structure encoded as well?